pkg/v2/tkr/resolver/internal/resolver_impl.go

```diff
@@ -0,0 +1,366 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package internal contains the implementation of the TKR Resolver.
+package internal
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/cluster-api/util/conditions"
+
+	runv1 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v2/tkr/resolver/data"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v2/tkr/util/version"
+)
+
+type Resolver struct {
+	cache cache
+}
+
+func NewResolver() *Resolver {
+	return &Resolver{cache: cache{
```

Comment:
The idea is that it should be irrelevant if Resolver is short or long lived. `cache` in the implementation is definitely out of the desire to reuse known TKRs and OSImages for more than one `Resolve()` call. 

Cluster webhook would list all TKRs and OSImages and populate the cache on initialization. It would then watch TKRs and OSImages and simply `Add()` incoming updates thereby providing background cache refresh and invalidation.

Reply:
Adding this explanation to the readme.